tests/conftest.py

```diff
@@ -33,7 +33,24 @@ def stampy_duplicates():
 
 
 @pytest.fixture
-def mock_retriever_model():
+def mock_pinecone_search():
+    def get_pinecode_results(xq, namespace, top_k=None, filter=None, **kwargs):
+        """There are a load of Pinecone results saved as json results, so just read from the appropriate one."""
+        with open(Path(__file__).parent / (f'data/pinecone/{xq}+{namespace}+{filter}+{top_k}.json')) as f:
+            data = json.load(f)
+
+        matches = [
+            Mock(id=item['id'], score=item.get('score'), metadata=item.get('metadata', item), __getitem__=getattr)
```

Comment:
a bit of magic here, so the results can be used both as `result.field` and `result["field"]`

Reply:
I noticed sometimes `result.field` was allowed but other times `result["field"]` was required. Must admit I didn't quite understand when or why.